bpf/lib/clustermesh.h

```diff
@@ -1,11 +1,48 @@
 /* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
 #ifndef _H_LIB_CLUSTERMESH_H_
 #define _H_LIB_CLUSTERMESH_H_
+#include "lib/utils.h"
+
+#define CLUSTER_ID_LOWER_MASK 0x000000FF
+
+#ifndef __CLUSTERMESH_HELPERS__
+#define __CLUSTERMESH_HELPERS__
+/* these macros allow us to override the values in tests */
+#define IDENTITY_LEN get_identity_len()
+#define IDENTITY_MAX get_max_identity()
+
+static __always_inline __u32
+get_identity_len()
+{
+	__u32 identity_len = CONFIG(IDENTITY_LEN);
```

Comment:
I'm slightly surprised this works, since `IDENTITY_LEN` is defined to `get_identity_len()` a few lines above. I think that's due to the CONFIG() macro pasting the argument verbatim during the initial preprocessor pass. I can see this confusing some editors/language servers though.

Since this is a datapath configurable, what about calling this `CONFIG(cluster_id_length)`? Uppercase names imply preprocessor directives, which IDENTITY_LEN technically isn't. When generating Go code, we can nicely transform `cluster_id_length` into `ClusterIDLength`. We're setting the convention for all future configurables here, so worth thinking this through.

Reply:
yep good point, that makes sense. i'll push that change

Reply:
updated :+1:

Reply:
Thanks, looks good!